### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by groutr**
Summary of changes:
* implement a much faster (and much more memory efficient) `ascii_image2`.
* Avoid building strings when lists/arrays will work.
* Avoid mutating list while iterating thru same list.
* Added an fps counter with `--fps` flag.

The result of this PR essentially doubles the framerate on my laptop (primarily because of the new `ascii_image2` implementation) from about 14fps to 30fps. It probably could go faster than 30fps, but my webcam only supports 30fps.

### CODE REVIEW

**Comment by groutr** [comment]
@joschuck I need to add that this has been very entertaining to play with!

**Comment by groutr** [comment]
The latest round of changes now gets processed at full framerate for my camera by resizing to target resolution before any processing. Preallocating arrays and reusing arrays also helps out.

**Comment by joschuck** [comment]
wow, I knew it was far for optimal but did not find the time, thanks for that! will have a look in a few hours and try to make a new release.

**Comment by groutr** [comment]
Looks like I left some camera specific property stuff on there. Let me get
that cleaned up.

**Comment by groutr** [comment]
This should be reviewable now. One thing I have noticed is the segmentation doesn't work well unless there is good lighting. I made sure to preserve enough resolution for segmentation. We could be smarter about resizing.

**Review by joschuck — APPROVED** [review]
lgtm, i will merge this into new development branch now and then correct for code formatting and mypy warnings.

File: matrix_webcam/__main__.py [hunk 1]
```diff
@@ -85,91 +97,112 @@ def main() -> None:
...
     with SelfieSegmentation(model_selection=1) as selfie_segmentation:
         while cap.isOpened():
-            success, image = cap.read()
+            success, image = cap.read(image)
 
             if not success:
                 print("Ignoring empty camera frame.")
-                # If loading a video, use 'break' instead of 'continue'.
                 continue
+            image = cv2.flip(image, 1, image)
```
Comment (joschuck) [inline matrix_webcam/__main__.py:133]: left hand assignment is not needed btw opencv, if you write it like this you see why cv2.flip(image, 1, dst=image) i think that happened somewhere down in the code too

Comment (groutr) [reply matrix_webcam/__main__.py:133]: I realize that, however, my stylistic choice is to have that assignment as I think it makes it easier to catch mistakes.

**Comment by ChrisCarini** [comment]
@joschuck - curious, is there a guesstimated ETA on when the development branch will be released to a new version? I am interested in leveraging the changes in this PR.

**Pull request merged**